exercises/python/python_for_probability_bonus_level.ipynb

```diff
@@ -0,0 +1,310 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "b) What is the probability that the sniper doesn't shoot a runner?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "c) What is the probability that the sniper shoots a giant and fails?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "d) If we notice that the sniper failed a shot, what is the probability that he was trying to shoot a walker?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Exercise 2: Time-travelling test subject [0.25 points]\n",
+    "\n",
+    "Let's say that we've developed a time-travelling watch, but it can only take you exactly 5 years into the future or 5 years back in time. Additionally, once you use it, it only allows you to go back to the present, or stay in whichever time you're now. Our secret organization is ready to run some tests on humans, so they are sending the first human subject. Now, while waiting for the test to start, one of your collegues starts talking to you. He's trying to guess what the human subject will do. He thinks that there is a 80% chance that the subject will travel to the future. And if the subject doesn't travel to the future, there is a 70% chance that he'll come back to the present. On the other hand, if the subject does travel to the future, there is a 60% chance that he'll come back to the present.\n",
+    "\n",
+    "a) Which interpretation of probability is being used here and why?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "b) What events are mentioned in the problem description and what are their probabilities? Which ones of these probabilities are priors and which ones are conditional probabilities?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "c) What is the probability that the subject never comes back?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "d) After the experiment, the subject didn't come back to the present. What is the probability that the subject travelled to the past?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Exercise 4: Joint Probability Distributions [0.25 points]\n",
```

Comment:
Exercise 4 is before exercise 3?

Reply:
you right! I changed the location of the last 2 problems and forgot to change the numbers.
Thanks